spec/mongo/transactions_examples_spec.rb

```diff
@@ -0,0 +1,236 @@
+require 'spec_helper'
+
+describe 'transactions examples in Ruby', if: test_transactions? do
+
+  let(:hr) do
+    authorized_client.use(:hr).database
+  end
+
+  let(:reporting) do
+    authorized_client.use(:reporting).database
+  end
+
+  before(:each) do
+    hr[:employees].insert_one(employee: 3, status: 'Active')
+    reporting[:events].insert_one(employee: 3, status: { new: 'Active', old: nil})
+  end
+
+  after(:each) do
+    hr.drop
+    reporting.drop
+  end
+
+  context 'individual examples' do
+
+    let(:session) do
+      authorized_client.start_session
+    end
+
+    # Start Transactions Intro Example 1
+
+    def update_employee_info(session)
+      employees_coll = session.client.use(:hr)[:employees]
+      events_coll = session.client.use(:reporting)[:events]
+
+      session.start_transaction(read_concern: { level: :snapshot },
+                                write_concern: { w: :majority })
+      employees_coll.update_one({ employee: 3 }, { '$set' => { status: 'Inactive'} },
+                                session: session)
+      events_coll.insert_one({ employee: 3, status: { new: 'Inactive', old: 'Active' } },
+                             session: session)
+
+      loop do
+        begin
+          session.commit_transaction
+          puts 'Transaction committed.'
+          break
+        rescue Mongo::Error::OperationFailure, Mongo::Error::SocketError,
+               Mongo::Error::NoServerAvailable => e
+
+          if e.label?(Mongo::Session::UNKNOWN_TRANSACTION_COMMIT_LABEL)
```

Comment:
How did UnknownTransactionCommitResult become UNKNOWN_TRANSACTION_COMMIT_LABEL, and why is it under Session?

Reply:
This is the constant the driver uses for this string internally, and it's provided publicly to allow users to use it specifically for this case. I think users should use our constants for labels rather than risk making a typo, so I used the constant here to demonstrate the label.

Reply:
The name of the constant does not match the documented name of the label ("result" part has gone away). Additionally, as a user it is not at all clear why this constant exists under Session. This is two cases of driver implementation details being exposed to the users. While specifying a literal string is somewhat error-prone, encouraging users to utilize driver guts like this to me is worse.

Reply:
The Java version has constants for labels and exposes them appropriately.

Reply:
The only time this label will ever exist is on errors returned by the commit_transaction method, so I don't see how namespacing the error label in this class exposes anything more than that. I don't see any issue with the way that the Python driver deals with this issue, and we're doing it the same way.